src/js/components/TextInput/TextInput.js

```diff
@@ -327,6 +327,8 @@ const TextInput = forwardRef(
           {...dropProps}
         >
           <ContainerBox
+            id={`${id}-listbox`}
```

Comment:
I think we should include these `id` elements conditionally, only if the caller provides an `id`, otherwise we are likely to be rendering multiple elements in the DOM with the same `id`. This would follow the pattern we have at line 321.

Reply:
Oh, that’s a good point. If the option IDs are omitted, then `aria-activedescendent` would have to be removed, which then means the active option can’t be announced by a screen reader.  

Similarly, a combobox must be linked to a listbox via `aria-controls`, so removing the listbox ID would also remove the combobox role from the text input. 

Basically, I think all the combobox props are conditional on IDs being defined.

Reply:
Is there any way that we could generate a unique ID by default for a component, if one hasn’t been passed in? ARIA pretty much relies on element IDs to link up related DOM nodes.

Reply:
We haven't gone down that path so far. And instead pursuing things like eslint plugins to warn when an `id` is missing, notably in `input` + `label` situations.

https://github.com/grommet/eslint-plugin-grommet

Reply:
Sounds good. I've added IDs to the storybook, and conditional checks so that the combobox role and states are only added to the text input if the component has both `id` and `suggestions`. I've added conditional checks to the `listbox` and `option` IDs too.